api/hasura/actions/_handlers/linkDiscordCircle.ts

```diff
@@ -0,0 +1,83 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { z } from 'zod';
+
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { errorResponseWithStatusCode } from '../../../../api-lib/HttpError';
+import { verifyHasuraRequestMiddleware } from '../../../../api-lib/validate';
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+} from '../../../../src/lib/zod';
+
+export const linkDiscordCircleInputSchema = z
+  .object({
+    circle_id: z.string(),
```

Comment:
Rather than casting this as a number below, zod provides [coercion facilities now](https://zod.dev/?id=coercion-for-primitives). Once my version bump gets merged, we can probably update this.